projects/igniteui-angular/src/lib/core/selection.ts

```diff
@@ -1,141 +1,98 @@
+        let selection: Set<any>;
+        itemIDs.forEach((item) => selection = this.select_item(componentID, item, selection, clearSelection));
         return selection;
     }
 
-    public append_items(componentID: string, itemIDs: any[]): any[] {
-        let selection = this.get_selection(componentID);
-        if (selection === undefined) {
-            selection = [];
-        }
-        return [...selection, ...itemIDs];
-    }
-
     /**
      * Remove item from component selection using item's ID.
      * @param componentID ID of the component, which we remove items from.
      * @param itemID ID of the item to remove from component selection.
-     * @param compareFunc Custom function that compares item to deselect with those presented in the current selection.
-     * If not presented, then the item equal by reference to any item from the selection (using ===) is removed.
-     * @param currSelection Used internally only by the selection (deselect_items method) to accumulate deselected items.
+     * @param sel Used internally only by the selection (deselect_items method) to accumulate deselected items.
      *
      * @returns selection after the item is removed
      */
-    public deselect_item(componentID: string, itemID, compareFunc?: (item) => boolean, currSelection?: any[]) {
-        if (!currSelection) {
-            currSelection = this.get_selection(componentID);
+    public deselect_item(componentID: string, itemID, sel?: Set<any>) {
+        if (!sel) {
+            sel = this.get_selection(componentID);
         }
-        if (currSelection === undefined) {
+        if (sel === undefined) {
             return;
         }
-        if (!compareFunc) {
-            return currSelection.filter((item) => item !== itemID);
-        } else {
-            return currSelection.filter(compareFunc);
-        }
+        sel.delete(itemID);
+        return sel;
     }
 
     /**
      * Remove items from component selection using their IDs.
      * @param componentID ID of the component, which we remove items from.
      * @param itemID ID of the items to remove from component selection.
-     * @param compareFunc Custom function that compares items to deselect with those presented in the current selection.
-     * If not presented, then all items equal by reference to any item from the selection (using ===) are removed.
      *
      * @returns selection after the item is removed
      */
-    public deselect_items(componentID: string, itemIDs: any[], compareFunc?: (item) => boolean, ): any[] {
-        let selection: any[];
-        itemIDs.forEach((deselectedItem) => selection = this.deselect_item(componentID, deselectedItem, compareFunc, selection));
+    public deselect_items(componentID: string, itemIDs: any[]): Set<any> {
+        let selection: Set<any>;
+        itemIDs.forEach((deselectedItem) => selection = this.deselect_item(componentID, deselectedItem, selection));
```

Comment:
This does not seems right 
In deselect_item, it will not enter the `!sel` *if*, skipping `sel = this.get_selection(componentID)`, so it will `delete` items from an empty set.